src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -156,7 +150,15 @@ private void EnableTimer(float pollingIntervalInSeconds)
                         restoreFlow = true;
                     }
 
-                    _pollingTimer = new Timer(s => ((CounterGroup)s!).OnTimer(null), this, _pollingIntervalInMilliseconds, _pollingIntervalInMilliseconds);
+                    if (_pollingThread == null)
+                    {
+                        _pollingThread = new Thread(() => PollForValues());
+                        _pollingThread.Start();    
```

Comment:
It may be a good idea to set the `_pollingThread` field only once the the thread starts successfully.